server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -306,9 +306,26 @@ public ClusterState execute(ClusterState currentState) {
                 ensureNoCleanupInProgress(currentState, repositoryName, snapshotName, "create snapshot");
                 ensureBelowConcurrencyLimit(repositoryName, snapshotName, snapshots, deletionsInProgress);
                 // Store newSnapshot here to be processed in clusterStateProcessed
-                List<String> indices = Arrays.stream(indexNameExpressionResolver.concreteIndexNames(currentState, request))
-                    .filter(indexName -> systemIndices.isSystemIndex(indexName) == false) // Only resolve system indices via Features
-                    .collect(Collectors.toList());
+                Map<Boolean, List<String>> requestedIndices = Arrays.stream(
+                    indexNameExpressionResolver.concreteIndexNames(currentState, request)
+                ).collect(Collectors.partitioningBy(systemIndices::isSystemIndex));
+
+                List<String> requestedSystemIndices = requestedIndices.get(true);
+                if (requestedSystemIndices.isEmpty() == false) {
+                    Set<String> explicitlyRequestedSystemIndices = new HashSet<>(requestedSystemIndices);
+                    explicitlyRequestedSystemIndices.retainAll(Arrays.asList(request.indices()));
+                    if (explicitlyRequestedSystemIndices.isEmpty() == false) {
+                        throw new IllegalArgumentException(
+                            new ParameterizedMessage(
+                                "system indices [{}] were explicitly requested, but snapshots should use the "
+                                    + "[include_global_state] or [feature_states] parameters to control inclusion of system indices",
```

Comment:
Not a blocker, but we may want to mention the `indices` parameter here. Right now, it can be a little unclear where the error is coming from. I'd suggest something like:

> The [indices] parameter includes system indices [{}]. To include system indices in a snapshot, use the [include_global_state] or [feature_states] parameters instead.

It'd be great if we could somehow surface this error for SLM policies too (either at policy creation/update or execution), but that's likely outside of scope here.

Reply:
I opened an issue for detecting this in SLM policies at https://github.com/elastic/elasticsearch/issues/81319.

I'll make the wording change to the message, thanks!